examples/enc-search-sample.rs

```diff
@@ -0,0 +1,427 @@
+    // end-snippet{createSecondIndex}
+
+    // Now save customers using both blind indices
+    let cust24 = Customer { id: 24, name: "Pokey".to_string(), email: "pokey@gumby.io".to_string() };
+    save_customer2(&cust24, &group_id, &sdk, &blind_index, &blind_index2).await?;
+
+    Ok(())
+}
+
+// Allow the user to enter additional customers and store each of them in the mock server
+async fn add_customers(
+    last_id: &u32,
+    group_id: &ironoxide::group::GroupId,
+    sdk: &ironoxide::IronOxide,
+    blind_index: &ironoxide::search::BlindIndexSearch) -> Result<()> {
+
+    let mut next_id = last_id + 1;
+    loop {
+        let mut name = String::new();
+        println!("Enter next customer name, or blank line to quit");
+        std::io::stdin()
+            .read_line(&mut name)
+            .expect("error: couldn't read input");
+        name = name.trim().to_string();
+        if &name == "" {
+            break;
+        } else {
+            let mut email = String::new();
+            println!("Enter customer's email");
+            std::io::stdin()
+                .read_line(&mut email)
+                .expect("error: couldn't read input");
+            email = email.trim().to_string();
+
+            let cust = Customer { id: next_id, name: name, email: email };
+            save_customer(&cust, &group_id, &sdk, &blind_index).await?;
+        }
+        next_id += 1;
+    }
+
+    Ok(())
+}
+
+async fn save_customer(
+    cust: &Customer,
+    group_id: &ironoxide::group::GroupId,
+    sdk: &ironoxide::IronOxide,
+    blind_index: &ironoxide::search::BlindIndexSearch) -> Result<()>
+{
+    // start-snippet{indexData}
+    // Generate tokens need to search for matching customer names
+    let name_tokens = blind_index
+        .tokenize_data(&cust.name, None)?
+        .into_iter()
+        .collect::<Vec<u32>>();
+
+    // Encrypt the name and email addresses to protect privacy. Also need to store EDEKs
+    // to decrypt them.
+    let encrypt_opts = DocumentEncryptOpts::with_explicit_grants(
+        None,                  // document ID - create unique
+        None,                  // document name
+        false,                 // don't encrypt to self
+        vec![group_id.into()], // users and groups to which to grant access
+    );
+    let enc_name = sdk
+        .document_encrypt_unmanaged(cust.name.as_bytes(), &encrypt_opts)
+        .await?;
+    let enc_email = sdk
+        .document_encrypt_unmanaged(cust.email.as_bytes(), &encrypt_opts)
+        .await?;
+
+    let enc_cust = EncryptedCustomer {
+        id : cust.id,
+        enc_name: enc_name.encrypted_data().to_vec(),
+        name_keys: enc_name.encrypted_deks().to_vec(),
+        enc_email: enc_email.encrypted_data().to_vec(),
+        email_keys: enc_email.encrypted_deks().to_vec()
+    };
+    save_customer_to_app_server(enc_cust, name_tokens, vec![]);
+    // end-snippet{indexData}
+
+    Ok(())
+}
+
+async fn save_customer2(
```

Comment:
It seems like we should comment why there are 2 save_customers and what is different about them. Alternatively, just give them function a more descriptive name.

Reply:
Done.